sys/arm64/arm64/hyp_stub.S

```diff
@@ -37,7 +37,12 @@
  */
 .macro vector_stub_el1h_sync
 	.align 7
+#if __has_feature(capabilities)
```

Comment:
Hmm, a capability is not present for purecap. We install the handlers like this `vmm_call_hyp0(vtophys(&vmm_hyp_code));`, and vtophys() doesn't yield a capability of course. So I don't quite understand how this works.

Reply:
We don't currently clear DDC_EL2, we only clear DDC_EL1 via clear_capregs clearing the current EL's DDC

Reply:
Oh humm.  Presumably you would want to save off the EL2 DDC in something hidden from EL1 when EL2 inits and then use that to derive a suitable cap in this function (along with clearing EL2 DDC presumably).  Not sure how much work that is though, maybe best to document it with an XXX for now?  (And maybe create an issue to track this?)

Reply:
I already need to create a root capability for EL2's address space, akin to `kernel_root_cap`. So I think I can clear DDC_EL2 in locore before dropping to EL1, then modify the vmm initialization routine to derive a capability from `vtophys(&vmm_hyp_code)`. At least, that's what I'm trying now.

Reply:
That sounds like what I'd suggest

Reply:
I ended up using `kernel_root_cap` directly. (My notion of using the EL2 root cap doesn't make sense since we're dealing with a physical address here, not an EL2 virtual address.) The updated patch series includes a new commit to clear DDC_EL2.

Reply:
Oh, worth pointing out: this is cvtp not cvtd so it uses PCC as the root. So this would actually have worked for a purecap kernel even with DDC_EL2 clear, but still not the best design.